frontend_tests/node_tests/markdown.js

```diff
@@ -229,8 +229,15 @@ var bugdown_data = JSON.parse(fs.readFileSync(path.join(__dirname, '../../zerver
          expected: '<p><img alt=":poop:" class="emoji" src="/static/generated/emoji/images/emoji/unicode/1f4a9.png" title="poop"></p>'},
         {input: '\u{1f937}',
          expected: '<p>\u{1f937}</p>' },
+        // Test only those realm filters which don't return True for
+        // `contains_backend_only_syntax()`. Those which return True
+        // are tested separately.
         {input: 'This is a realm filter #1234 with text after it',
          expected: '<p>This is a realm filter <a href="https://trac.zulip.net/ticket/1234" target="_blank" title="https://trac.zulip.net/ticket/1234">#1234</a> with text after it</p>'},
+        {input: '#1234is not a realm filter.',
+         expected: '<p>#1234is not a realm filter.</p>'},
+        {input: 'A pattern written as #1234is not a realm filter.',
+         expected: '<p>A pattern written as #1234is not a realm filter.</p>'},
```

Comment:
For doing so, we will need to ensure same realm filter is configured in test suites both on frontend as well as backend.